mantidproject/mantid
# Use highestVersion instead of getDescriptors for script histories

**Description of work.**
Adds a method get a single descriptor in the Algorithm factory, this avoids us having to get all algorithm descriptors each time we want a descriptor

Previously every algorithm object was allocated (900+) then discarded for each workspace. These changes drop the project saving time from 2+ minutes to ~5 seconds in debug mode for a large case.

**To test:**
- Build master copy of Mantid
- Run the following script
```
for i in range(100):
    ws_name = str(i)
    CreateSampleWorkspace(OutputWorkspace=ws_name)
    for i in range(100):
        AddSampleLog(ws_name,'history_destroyer','blah')
```
- Wait for project recovery to start saving after the script has run (set logger to debug)
- Wait several minutes for it to save

- Build these changes
- Repeat and verify the time taken.



Fixes #22788 . 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
